### DISCUSSION TITLE
Get Output of previous Action (Docker)

### DISCUSSION THREAD

**Discussion by KodaCHC** [community/community#25278] [Actions]
<p>I have a workflow with multiple jobs. One of them is a Github Action as a Docker container.</p>
<p>Here I can pass a value to my action without problems</p>
<pre><code class="lang-auto">- name: Run my Docker Action
uses: ./.github/actions/MyDockerAction
env:
  MY_VAR1: ${{ steps.getVar.outputs.data }}
</code></pre>
<p>But what is the correct way if I want to use a variable from my Docker action in the next job?<br>
I tried to work with export outvar1=1 in entrypoint.sh and use ${{ env.outvar1 }} in the next step. But this variable remains empty</p>

**Comment by airtower-luna** [comment]
<p>Set an <em>output</em> inside the action, give the step an ID, and then you can use the output in the following steps.</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/5814919/181090047-f02e2d53-7202-4293-ab1d-418d57089cca.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter" target="_blank" rel="noopener">Workflow commands for GitHub Actions - GitHub Docs</a></h3>

  <p>You can use workflow commands when running shell commands in a workflow or in an action's code.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by KodaCHC** [comment]
<p>Thank you. I have found this, but have not checked it since uses and run can not be used together?</p>
<p>This leads to at least one error:</p>
<blockquote>
<ul>
<li>name: Run my Docker Action<br>
uses: ./.github/actions/MyDockerAction<br>
id: myID<br>
run: |<br>
echo “::set-output name=outvar1::$(outvar1)\n”<br>
env:<br>
MY_VAR1: ${{ steps.getVar.outputs.data }}</li>
</ul>
</blockquote>

**Comment by airtower-luna** [comment]
<p>You can’t combine <code>uses</code> and <code>run</code> in one step. Your step must use exactly one of those. The output command would need to be written by the entrypoint script of your Docker action.</p>

**Comment by KodaCHC** [comment]
<p>Thanks. I thought I had done that with “export outvar1=1”. I have to search again because I’m not sure what I just misunderstood.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><code>export</code> is a shell built-in that sets a variable to be exported (made available) to subsequent comments run <em>from that shell</em>. The shell running in your step container exits with the step, and its status is lost.</p>
<p>If you want to transmit any data to later steps you have to either use the methods provided by the Actions runner (set an output, or define an environment variable via <code>GITHUB_ENV</code>), or write to the file system.</p>
